src/components/data/utils.js

```diff
@@ -18,4 +21,31 @@ function getEncodedPath(path) {
     return encodedPath;
 }
 
-export { getEncodedPath };
+const validateDiskResourceName = (name) => {
```

Comment:
I stole this from @psarando App Launch code.  If App Launch gets merged first, I'll refactor to use that